From a9ba3ceb7dfbf84066f906ba6ee3c794bca6c7f5 Mon Sep 17 00:00:00 2001
From: Chad Dupuis <cdupuis@redhat.com>
Date: Mon, 18 Aug 2014 19:56:19 -0400
Subject: [scsi] qla2xxx: Enable the Flash Access Control (FAC) mailbox command

Message-id: <1408391817-6389-32-git-send-email-cdupuis@redhat.com>
Patchwork-id: 87920
O-Subject: [PATCH RHEL 7.1 e-stor 31/69] qla2xxx: Enable the Flash Access Control (FAC) mailbox command.
Bugzilla: 1089346
RH-Acked-by: Rob Evers <revers@redhat.com>
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>

From: Joe Carnuccio <joe.carnuccio@qlogic.com>

scsi commit id 1ca60e3b0dcbf1ab2be77eeaf50b048c6e9fd62f

Signed-off-by: Joe Carnuccio <joe.carnuccio@qlogic.com>
Signed-off-by: Saurav Kashyap <saurav.kashyap@qlogic.com>
Signed-off-by: James Bottomley <JBottomley@Parallels.com>
Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/drivers/scsi/qla2xxx/qla_init.c b/drivers/scsi/qla2xxx/qla_init.c
index b3f124f..1344ef4 100644
--- a/drivers/scsi/qla2xxx/qla_init.c
+++ b/drivers/scsi/qla2xxx/qla_init.c
@@ -1718,9 +1718,6 @@ enable_82xx_npiv:
   spin_unlock_irqrestore(&ha->hardware_lock, flags);
  }
 
- if (IS_QLA83XX(ha))
-  goto skip_fac_check;
-
  if (rval == QLA_SUCCESS && IS_FAC_REQUIRED(ha)) {
   uint32_t size;
 
@@ -1733,7 +1730,7 @@ enable_82xx_npiv:
        "Unsupported FAC firmware (%d.%02d.%02d).\n",
        ha->fw_major_version, ha->fw_minor_version,
        ha->fw_subminor_version);
-skip_fac_check:
+
    if (IS_QLA83XX(ha)) {
     ha->flags.fac_supported = 0;
     rval = QLA_SUCCESS;
-- 
1.7.1